main.go

```diff
@@ -230,7 +234,17 @@ func main() {
 	done := make(chan os.Signal, 1)
 	signal.Notify(done, syscall.SIGINT, syscall.SIGTERM)
 
-	ctx, cancel := context.WithCancel(context.Background())
+	var (
+		ctx    context.Context
+		cancel context.CancelFunc
+	)
+
+	if maxRuntime <= 0 {
+		ctx, cancel = context.WithCancel(context.Background())
+	} else {
+		ctx, cancel = context.WithTimeout(context.Background(), time.Duration(maxRuntime))
+	}
```

Comment:
To avoid the if-else we could also keep wrapping them:

```go
	ctx, cancel := context.WithCancel(context.Background())
	defer cancel()

	if maxRuntime > -1 {
		ctx, cancel = context.WithTimeout(ctx, maxRuntime)
		defer cancel()
	}
```